server/src/printers/NativeMakefilePrinter.cpp

```diff
@@ -42,7 +42,15 @@ namespace printer {
             argument = "";
         }
     }
+
+    static void eraseIfWlaAsNeeded(string &argument) {
+        if (argument == "-Wl,--as-needed") {
+            argument = "";
+        }
+    }
+
     static void removeLinkerFlag(string &argument, string const &flag) {
+        eraseIfWlaAsNeeded(argument);
```

Comment:
What might cause this change? In any case, you can use `removeLinkerFlag(argument, "--as-needed")` from the outside

Reply:
with this option left run tests fails with: `/usr/bin/ld: unrecognized option '-Wl,--as-needed'`

Reply:
changed so that this option is removed only for `/usr/bin/ld` in [5c9a2bb](https://github.com/UnitTestBot/UTBotCpp/pull/184/commits/5c9a2bb6547684d187e585187d11ae7e4cc81c49)

Reply:
Ah, I see. Look, `-Wl,option` is used by the compiler in order to pass `option` to the linker. So, in other words, the following commands are nearly the same:
- `clang -Wl,--as-needed`
- `ld --as-needed`
I think it may make sense to just remove `-Wl,` prefix when changing compiler to `ld` in your case. But frankly, I don't know what's the reason to do that.

Reply:
I see. Just changed the prefix for now then. I heard there was some project for which the change to use `ld` was introduced but don't know the details either.

Reply:
Nice to see, approved